start

```diff
@@ -498,10 +498,19 @@ function upgrade_local() {
     echo "upgrades: protocolversion done"
   fi
 
-  # Upgrade local network's soroban config to match testnet, unless the limits
+  # Upgrade local network's soroban config to match the phase1 file in stellar-core, unless the limits
   # have been configured with 'default', which will cause the limits to be left
   # in their default state.
   if [ $PROTOCOL_VERSION -ge 20 ] && [ "$LIMITS" != "default" ]; then
+    local SETTINGS_FILE
+    if [ "$LIMITS" = "testnet" ]; then
+      local CORE_VERSION=$(stellar-core version | head -n 1)
+      curl -O "https://raw.githubusercontent.com/stellar/stellar-core/$CORE_VERSION/soroban-settings/phase1.json"
```

Comment:
I think you need some more options on that curl command otherwise it'll fail silently and write garbage into the json file. I suggest at least adding:
- `f` - Fail on non 2xx status codes.
- `L` - Follow redirects.
- `s` - Don't output the progress bar which won't mean much to someone watching it.

```suggestion
      curl -fLOs "https://raw.githubusercontent.com/stellar/stellar-core/$CORE_VERSION/soroban-settings/phase1.json"
```